src/client/db.rs

```diff
@@ -0,0 +1,109 @@
+use bson;
+use bson::Bson;
+use client::MongoClient;
+use client::coll::Collection;
+use client::common::{ReadPreference, WriteConcern};
+
+/// Interfaces with a MongoDB database.
+pub struct Database<'a> {
+    pub name: String,
+    pub client: &'a MongoClient,
+    read_preference: Option<ReadPreference>,
```

Comment:
Alright. But we should allow also specifying the `write_concern` on a per-operation basis too. Remember to set `read_preference` and `write_concern` to their defaults. `read_preference`, pretty sure, defaults to `PRIMARY`.